Infrastructure/Resources/New-StorageAccountContainer.ps1

```diff
@@ -0,0 +1,77 @@
+<#
+
+.SYNOPSIS
+Create an ARM Storage Container.
+
+.DESCRIPTION
+Create one or more containers in a storage account
+
+.PARAMETER Location
+The location of the Resource Group. This is limited to West Europe and North Europe
+
+.PARAMETER Name
+The name of the Storage Account
+
+.PARAMETER ContainerName
+The names of one or more Containers to create in the Storage Account
+
+.EXAMPLE
+.\New-StorageAccountContainer.ps1 -Location "West Europe" -Name stracc -ContainerName public
+
+.EXAMPLE
+.\New-StorageAccountContainer.ps1 -Location "West Europe" -Name stracc -ContainerName public,private,images
+
+#>
+
+Param(
+    [Parameter(Mandatory = $false)]
+    [ValidateSet("West Europe", "North Europe")]
+    [String]$Location = $ENV:Location,
+    [Parameter(Mandatory = $true)]
+    [String]$Name,
+    [Parameter(Mandatory = $true)]
+    [String[]]$ContainerName
```

Comment:
If it's an array should it be ContainerNames?

Reply:
I usually follow these guidelines:

```
Use Singular Parameter Names
Avoid using plural names for parameters whose value is a single element. This includes parameters that take arrays or lists because the user might supply an array or list with only one element.
```

https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx